src/test/ui/iterators/invalid-iterator-chain.stderr

```diff
@@ -52,14 +52,14 @@ LL |             .sum::<i32>(),
              <i32 as Sum<&'a i32>>
              <i32 as Sum>
 note: the method call chain might not have had the expected associated types
-  --> $DIR/invalid-iterator-chain.rs:20:14
+  --> $DIR/invalid-iterator-chain.rs:25:14
    |
 LL |         vec![0, 1]
    |         ---------- this expression has type `Vec<{integer}>`
 LL |             .iter()
    |              ------ `Iterator::Item` is `&{integer}` here
 LL |             .map(|x| x * 2)
-   |              ^^^^^^^^^^^^^^ `Iterator::Item` changed to `{integer}` here
+   |              -------------- `Iterator::Item` changed to `{integer}` here
```

Comment:
This span (correctly) shouldn't be a primary span, I think.

Reply:
The idea behind these where "this is where the type was correct, and this is where it wasn't" for the primary spans, but it works either way.

Reply:
Yeah, but I'm not sure if I understand why it was counting this as an incorrect span in the first place 😅